polkadot/runtime/test-runtime/src/lib.rs

```diff
@@ -352,7 +353,8 @@ impl pallet_staking::Config for Runtime {
 	type SlashDeferDuration = SlashDeferDuration;
 	type AdminOrigin = frame_system::EnsureNever<()>;
 	type SessionInterface = Self;
-	type EraPayout = pallet_staking::ConvertCurve<RewardCurve>;
+	type MaxValidatorsPayout = MaxValidatorsPayout;
```

Comment:
```suggestion
```

If you were to do this approach, I don't think there is any need to pass `MaxValidatorsPayout` to the staking pallet. Whatever cap is needed, should already be applied to `RewardCurve` or whatever type is implementing `type EraPayout`.